### ISSUE TITLE
csv: Fix reading empty columns as float - [merged]

### ISSUE THREAD

**Issue by SGeeversAtVortech** [Deltares/rtc-tools#1455]
In GitLab by @vreeken on Dec 17, 2019, 13:17

_Merges csv-fix-emtpy-column-read-nan-try-2 -> master_

We use np.genfromtxt with dtype=None, which means that it will guess. If
a column does not have any value specified, it will guess the dtype
boolean with values `False`. This would eventually be turned into a
Timeseries of `0.0`. We however want a Timeseries full of `np.nan`.

It is difficult to make np.genfromtxt read in floats, when the first
column can sometimes be a string. Therefore, we detect the case where
the dtype of a column is boolean, and force a conversion to float.
Basically, we therefore only support reading in floats (and a optionally
string for the first column).

Closes #1128

**Comment by SGeeversAtVortech** [comment]
In GitLab by @vreeken on Dec 17, 2019, 13:52

added 1 commit

<ul><li>7e627bd8 - csv: Fix reading empty columns as float</li></ul>

[Compare with previous version](/deltares/rtc-tools/merge_requests/357/diffs?diff_id=67975361&start_sha=819836c36bb8208001aa1966824f98382d2a68e8)

**Comment by SGeeversAtVortech** [comment]
In GitLab by @vreeken on Dec 17, 2019, 13:54

added 1 commit

<ul><li>441d6ff6 - csv: Fix reading empty columns as float</li></ul>

[Compare with previous version](/deltares/rtc-tools/merge_requests/357/diffs?diff_id=67975728&start_sha=7e627bd8b164b71efd383c70d0bd42a530f71bdf)

**Comment by SGeeversAtVortech** [comment]
In GitLab by @vreeken on Jan 20, 2020, 13:51

added 2 commits

<ul><li>0ee47a74 - 1 commit from branch <code>master</code></li><li>4b770409 - csv: Fix reading empty columns as float</li></ul>

[Compare with previous version](/deltares/rtc-tools/merge_requests/357/diffs?diff_id=71689293&start_sha=441d6ff60aebf5dcd81336eb748cc26afa8ce13a)

**Comment by SGeeversAtVortech** [comment]
In GitLab by @vreeken on Jan 20, 2020, 13:52

enabled an automatic merge when the pipeline for 4b770409218e7bee232eb80e38a9ed811446f6cf succeeds

**Comment by SGeeversAtVortech** [comment]
In GitLab by @vreeken on Jan 20, 2020, 13:57

merged
